### DISCUSSION TITLE
Component with lots of telemetry

### DISCUSSION THREAD

**Discussion by rawinza555** [nasa/fprime#1086] [General]
So there's a component in my project that have to handle the interface to power control unit. Our PCU has 23 output channels and have a built in BMS that monitor 15 battery cells and my team want to monitor pretty much all of them. I could either have like 50+ telemetry channels (and adjust the based id window accordingly) or only two or three of them and serialize them together and let ground control software handle the serialization and the display. 

I'm wondering if having 50+ telemetry channels would have caused issues on the flight control software itself? Or is it a good practice to do this or serialize them so I don't need to have that much of the telemetry channels.

Almost forgot, each telemetry need to be in 32 bits float. I'm not sure if there is a way to implement this as a a part of a loop. something like
```
for (int i = 1; i <= MAX_CHANNEL_No;i++){
   U8 txData[7];
   txData[0] = READ_HEADER;
   txData[1] = CHANNEL_VOLTAGE_REGISTRY
   txData[2] = i //channel No.
   this->spiwriteread(0,tx,rx)
   something here // write a telemetry in channel i
}
```

**Comment by rawinza555** [comment]
Anyone have an input on this?

**Reply by SterlingPeet** [reply]
I don't have any reason to believe that the number of channels has any appreciable effect on the flight software framework.  However, in the interest of minimizing the code on the spacecraft, I usually don't deserialize subsystem telemetry that won't be used somewhere in the flight software.

In other words, if the onboard software needs to look at some channels (e.g. battery voltage) to determine the need to change modes, I deserialize and telemeter them.  Everything else, just gets packaged and sent to the ground as one large chunk of data.  From there, the operators can do whatever they want with it.

**Reply by rawinza555** [reply]
Thanks! I guess serializing them all would be the best option in my scenario. The next step would probably be making sure that the transceiver is able to handle this amount of bytes per transaction.

For mode selection, I opt for creating a port that PCU interface would send it to mode handling.

**Reply by SterlingPeet** [reply]
If the serialized data is too large, you can chunk it into smaller pieces to downlink, as long as you label them for reassembly.  If you cleverly choose the chunk sizes to occur at natural data boundaries in the serialized data, then each chunk can be deserialized independently by the operators.
